packages/patterns/src/components/partial-syncing-controls.js

```diff
@@ -42,15 +40,11 @@ function PartialSyncingControls( { name, attributes, setAttributes } ) {
 			return;
 		}
 
-		const updatedConnections = {
-			...attributes.connections,
-			attributes: {
-				...attributes.connections?.attributes,
-				[ attributeName ]: {
-					source: 'pattern_attributes',
-				},
-			},
-		};
+		for ( const attributeName of Object.keys( syncedAttributes ) ) {
+			updatedConnections.attributes[ attributeName ] = {
+				source: 'pattern_attributes',
+			};
+		}
```

Comment:
Similarly, when we enable pattern connections, we may want to check there's no other type of connection configured and ensure we don't overwrite it.